packages/svelte/src/internal/client/dom/blocks/html.js

```diff
@@ -46,15 +46,20 @@ export function html(node, get_value, svg, mathml, skip_warning) {
 
 	var value = '';
 
-	/** @type {Effect | null} */
+	/** @type {Effect | undefined} */
 	var effect;
 
 	block(() => {
-		if (value === (value = get_value())) return;
+		if (value === (value = get_value())) {
+			if (hydrating) {
+				hydrate_next();
+			}
+			return;
+		}
 
 		if (effect) {
```

Comment:
since we do this in other places...

```suggestion
		if (effect !== undefined) {
```